src/main/kotlin/no/skatteetaten/aurora/boober/service/EncryptionService.kt

```diff
@@ -60,3 +52,90 @@ class EncryptionService(
         }
     }
 }
+
+@Component
+class EncryptorWrapper(
+    @Value("\${boober.encrypt.key}") val key: String,
+) {
+    private val cipherThreadLocal: ThreadLocal<Cipher> = ThreadLocal()
+    private val providerName: String
+    private val strongKey: Key
+    val iterationCount = 65536
+
+    init {
+        val bouncyCastleProvider = org.bouncycastle.jce.provider.BouncyCastleProvider()
+        providerName = bouncyCastleProvider.name
+        Security.getProvider(providerName) ?: Security.addProvider(bouncyCastleProvider)
+        strongKey = keyFromPassword(key.toCharArray())
+    }
+
+    fun encrypt(message: ByteArray): ByteArray = runCipher(CipherMode.ENCRYPT_MODE, message)
+    fun decrypt(message: ByteArray): ByteArray = runCipher(CipherMode.DECRYPT_MODE, message)
+
+    private fun keyFromPassword(password: CharArray?): Key {
+        val spec: KeySpec = PBEKeySpec(password, salt, iterationCount, 128)
+        val factory = SecretKeyFactory.getInstance("PBKDF2WithHmacSHA256")
+        val tmp = factory.generateSecret(spec)
+
+        return SecretKeySpec(tmp.encoded, "AES")
+    }
+
+    private fun runCipher(cipherMode: CipherMode, message: ByteArray): ByteArray {
+        val mode = when (cipherMode) {
+            CipherMode.ENCRYPT_MODE -> Cipher.ENCRYPT_MODE
+            CipherMode.DECRYPT_MODE -> Cipher.DECRYPT_MODE
+        }
+        val cipher = getCipher()
+
+        cipher.init(mode, strongKey)
+        return cipher.doFinal(message) ?: throw IllegalStateException("Could not encrypt message")
+    }
+
+    private fun getCipher(): Cipher = cipherThreadLocal.get() ?: createAndSetCipher()
+
+    private fun createAndSetCipher(): Cipher {
+        val cipher = Cipher.getInstance("AES", providerName) ?: throw IllegalStateException("Could not find cipher")
+
+        cipherThreadLocal.set(cipher)
+        return cipher
+    }
+}
+
+private val salt = byteArrayOf(
```

Comment:
Ktlint vil ha den formatert som dette. Denne er tatt fra encryptor4j. Jeg har testet med genererte salter, men det må være denne

Reply:
Er det mulig å deaktivere ktlint regelen for dette arrayet?

Reply:
Jeg fikk tvingt den til å ikke wrappe nå :)